Src/SemanticComparison/ProxyGenerator.cs

```diff
@@ -173,13 +183,10 @@ private static void BuildMethodGetHashCode<TDestination>(TypeBuilder type)
             method.SetReturnType(typeof(int));
 
             int derivedGetHashCode = 135;
-            MethodInfo getHashCode = typeof(TDestination).GetMethod(

```

Comment:
>Could we trust tests in this project?

Well, you've also changed the tests so it's hard to tell...

I can imagine that you changed the tests because the `OperatingSystem` that we're passing in to the existing tests isn't available(?) Probably we can create our own `OperatingSystem` class though and keep the changes to the minimum?